### DISCUSSION TITLE
how to upload translation to "www.oqtane.net"

### DISCUSSION THREAD

**Discussion by ijaz-saeed** [oqtane/oqtane.framework#2212] [Q&A] [answered]
hi, 
when I search on "www.oqtane.net" for translations, I only see one entry. How can I upload a new Language translation to "www.oqtane.net"? 

thanks

**Comment by sbwalker** [comment, marked as answer]
@ijaz-saeed please send an email to support@oqtane.org and request an account be created. This will allow you to add your translation to the Oqtane Registry.

**Comment by sbwalker** [comment]
Instructions have now been added to https://www.oqtane.net/registry - I plan to enable Github Login shortly so that users do not need to have local login accounts.

**Reply by ijaz-saeed** [reply]
thanks @sbwalker for the help, but I am thinking why we need to upload meta-data on oqtane.net, if real source of the packages, is Nuget ?
are we not duplicating same information.

**Reply by sbwalker** [reply]
Nuget.org was never designed for third party platforms like Oqtane. There is no way to categorize packages in a granular way, no way to add additional custom metadata, and no way to manage version dependencies with the Oqtane framework itself. 

Technically speaking the Oqtane registry does not need to have a dependency on Nuget.org - however Nuget.org does offer some valuable benefits in terms of package validation, package analysis to prevent malicious code, and it is already highly trusted in the .NET ecosystem. 

But yes there is some duplicate effort for developers to list the packages in both places.

**Comment by sbwalker** [comment]
@ijaz-saeed I have been thinking more about the current dependency on Nuget.org for packages. Since most extensions such as translations are open source, I think it makes sense to also allow developers to link to packages which are published on Github. This would eliminate the need to manage the package in so many places.

**Reply by ijaz-saeed** [reply]
thanks @sbwalker for the clarification. and one more thing, I have sent a couple of emails to [support@oqtane.org](mailto:support@oqtane.org) but with no response. please if you can check, if these are not going to spam etc. thanks

**Reply by sbwalker** [reply]
Strange... yes I will check the spam folder.

**Reply by sbwalker** [reply]
I found your email in the spam folder... I will respond now

**Comment by sbwalker** [comment]
@ijaz-saeed the Oqtane Registry has now been updated to allow for GitHub login using OAuth2. In addition the Registry has been updated to allow packages to be hosted on Nuget.org or Github.com.

**Reply by ijaz-saeed** [reply]
thanks @sbwalker and I am finally able to upload my translation package to registry.

**Reply by sbwalker** [reply]
@ijaz-saeed I apologize that it took so long to get this configured. One suggestion is that you could change the name of your product to something more user friendly - similar to what Santiago did with his translation. Using the GitHub package name is technically correct but not very intuitive for users. @leigh-pointer should do the same.

**Reply by leigh-pointer** [reply]
Changed ;)

**Reply by ijaz-saeed** [reply]
Changed ...

**Comment by ijaz-saeed** [comment]
hi, @sbwalker I recently added newer version of my translation.  but it is not reflecting in the search results. any idea why it is not updating, 

here is the edit product screenshot 
![image](https://user-images.githubusercontent.com/21291112/188272223-4741ab13-e7ab-4d52-8d37-d17b6fa85f64.png)

here are the search results, 
![image](https://user-images.githubusercontent.com/21291112/188272249-14f46f64-484b-4ab7-b657-6bcce354185d.png)

to troubleshoot the issue, I de-activated the older version and only activated the new version. but no success. then I de-activated the product and tried to create a new product with new release version. again failure. and it seems new product registration is not working, is it so?

**Reply by sbwalker** [reply]
Please see #2323 - naming convention for framework translations needs to be Oqtane.Framework.{language}
